roachpb/data.go

```diff
@@ -784,6 +784,11 @@ func (s Span) Equal(o Span) bool {
 	return bytes.Equal(s.Key, o.Key) && bytes.Equal(s.EndKey, o.EndKey)
 }
 
+// ContainsKey returns whether this span contains the specified key.
+func (s Span) ContainsKey(key Key) bool {
```

Comment:
This is used from `engine.IsValidSplitKey` which was previously comparing on keys directly. Happy to move the comparisons back to that routine if it is too dangerous to leave here. But I'm not clear on how I can operate on `RKeys` at that level.